### PR TITLE
fixes partion name for devices like /dev/loop0

### PR DESCRIPTION
**Pull request by osixia**
Related issue #711

### CODE REVIEW

**Comment by guits** [comment]
jenkins test this please

**Comment by guits** [comment]
jenkins test this please

File: ceph-releases/jewel/ubuntu/14.04/daemon/osd_scenarios/osd_disk_activate.sh [hunk 1]
```diff
@@ -8,8 +8,8 @@ function osd_activate {
   fi
 
   CEPH_DISK_OPTIONS=()
-  DATA_UUID=$(blkid -o value -s PARTUUID "${OSD_DEVICE}"1)
-  LOCKBOX_UUID=$(blkid -o value -s PARTUUID "${OSD_DEVICE}"3 || true)
+  DATA_UUID=$(blkid -o value -s PARTUUID $(dev_part "${OSD_DEVICE}" 1))
```
Comment (leseb) [inline ceph-releases/jewel/ubuntu/14.04/daemon/osd_scenarios/osd_disk_activate.sh]: please quote like this `"$(dev_part "${OSD_DEVICE}" 1)"`

File: ceph-releases/jewel/ubuntu/14.04/daemon/osd_scenarios/osd_disk_activate.sh [hunk 2]
```diff
@@ -8,8 +8,8 @@ function osd_activate {
...
+  LOCKBOX_UUID=$(blkid -o value -s PARTUUID $(dev_part "${OSD_DEVICE}" 3) || true)
```
Comment (leseb) [inline ceph-releases/jewel/ubuntu/14.04/daemon/osd_scenarios/osd_disk_activate.sh]: ditto

File: ceph-releases/jewel/ubuntu/14.04/daemon/osd_scenarios/osd_disk_prepare.sh [hunk 3]
```diff
@@ -65,7 +65,7 @@ function osd_disk_prepare {
     echo "Unmounting LOCKBOX directory"
     # NOTE(leseb): adding || true so when this bug will be fixed the entrypoint will not fail
     # Ceph bug tracker: http://tracker.ceph.com/issues/18944
-    DATA_UUID=$(blkid -o value -s PARTUUID "${OSD_DEVICE}"1)
+    DATA_UUID=$(blkid -o value -s PARTUUID $(dev_part "${OSD_DEVICE}" 1))
```
Comment (leseb) [inline ceph-releases/jewel/ubuntu/14.04/daemon/osd_scenarios/osd_disk_prepare.sh]: ditto

File: ceph-releases/kraken/ubuntu/16.04/daemon/osd_scenarios/osd_disk_activate.sh [hunk 4]
```diff
@@ -8,8 +8,8 @@ function osd_activate {
   fi
 
   CEPH_DISK_OPTIONS=()
-  DATA_UUID=$(blkid -o value -s PARTUUID "${OSD_DEVICE}"1)
-  LOCKBOX_UUID=$(blkid -o value -s PARTUUID "${OSD_DEVICE}"3 || true)
+  DATA_UUID=$(blkid -o value -s PARTUUID $(dev_part "${OSD_DEVICE}" 1))
```
Comment (leseb) [inline ceph-releases/kraken/ubuntu/16.04/daemon/osd_scenarios/osd_disk_activate.sh]: ditto

File: ceph-releases/kraken/ubuntu/16.04/daemon/osd_scenarios/osd_disk_activate.sh [hunk 5]
```diff
@@ -8,8 +8,8 @@ function osd_activate {
...
+  LOCKBOX_UUID=$(blkid -o value -s PARTUUID $(dev_part "${OSD_DEVICE}" 3) || true)
```
Comment (leseb) [inline ceph-releases/kraken/ubuntu/16.04/daemon/osd_scenarios/osd_disk_activate.sh]: ditto

File: ceph-releases/kraken/ubuntu/16.04/daemon/osd_scenarios/osd_disk_prepare.sh [hunk 6]
```diff
@@ -65,7 +65,7 @@ function osd_disk_prepare {
     echo "Unmounting LOCKBOX directory"
     # NOTE(leseb): adding || true so when this bug will be fixed the entrypoint will not fail
     # Ceph bug tracker: http://tracker.ceph.com/issues/18944
-    DATA_UUID=$(blkid -o value -s PARTUUID "${OSD_DEVICE}"1)
+    DATA_UUID=$(blkid -o value -s PARTUUID $(dev_part "${OSD_DEVICE}" 1))
```
Comment (leseb) [inline ceph-releases/kraken/ubuntu/16.04/daemon/osd_scenarios/osd_disk_prepare.sh]: ditto

File: ceph-releases/luminous/ubuntu/16.04/daemon/osd_scenarios/osd_disk_prepare.sh [hunk 7]
```diff
@@ -66,7 +66,7 @@ function osd_disk_prepare {
     echo "Unmounting LOCKBOX directory"
     # NOTE(leseb): adding || true so when this bug will be fixed the entrypoint will not fail
     # Ceph bug tracker: http://tracker.ceph.com/issues/18944
-    DATA_UUID=$(blkid -o value -s PARTUUID "${OSD_DEVICE}"1)
+    DATA_UUID=$(blkid -o value -s PARTUUID $(dev_part "${OSD_DEVICE}" 1))
```
Comment (leseb) [inline ceph-releases/luminous/ubuntu/16.04/daemon/osd_scenarios/osd_disk_prepare.sh]: ditto

File: ceph-releases/luminous/ubuntu/16.04/daemon/osd_scenarios/osd_disk_activate.sh [hunk 8]
```diff
@@ -8,15 +8,15 @@ function osd_activate {
...
+  DATA_UUID=$(get_part_uuid $(dev_part "${OSD_DEVICE}" 1))
   if [[ -n "${OSD_JOURNAL}" ]]; then
     CLI+=("${OSD_JOURNAL}")
   else
     CLI+=("${OSD_DEVICE}")
   fi
   JOURNAL_PART=$(ceph-disk list "${CLI[@]}" | awk '/ceph journal/ {print $1}') # This is privileged container so 'ceph-disk list' works
   JOURNAL_UUID=$(get_part_uuid "${JOURNAL_PART}" || true)
-  LOCKBOX_UUID=$(get_part_uuid "${OSD_DEVICE}"3 || true)
+  LOCKBOX_UUID=$(get_part_uuid $(dev_part "${OSD_DEVICE}" 3) || true)
```
Comment (leseb) [inline ceph-releases/luminous/ubuntu/16.04/daemon/osd_scenarios/osd_disk_activate.sh]: ditto

File: ceph-releases/luminous/ubuntu/16.04/daemon/osd_scenarios/osd_disk_activate.sh [hunk 9]
```diff
@@ -8,15 +8,15 @@ function osd_activate {
   fi
 
   CEPH_DISK_OPTIONS=()
-  DATA_UUID=$(get_part_uuid "${OSD_DEVICE}"1)
+  DATA_UUID=$(get_part_uuid $(dev_part "${OSD_DEVICE}" 1))
```
Comment (leseb) [inline ceph-releases/luminous/ubuntu/16.04/daemon/osd_scenarios/osd_disk_activate.sh]: ditto

**Comment by leseb** [comment]
@osixia up!

**Comment by leseb** [comment]
jenkins test ceph_ansible-luminous-centos7-cluster

**Pull request merged**